src/ch/ch_monitor.c

```diff
@@ -913,18 +917,40 @@ static int virCHMonitorReadProcessEvents(virCHMonitorPtr mon)
     return (events > 0) ? virCHMonitorProcessEvents(mon, events) : events;
 }
 
+#define MONITOR_TMP_FAIL_RETRIES 5
+
 static void virCHMonitorEventLoop(void *data)
 {
     virCHMonitorPtr mon = (virCHMonitorPtr)data;
-    virDomainObjPtr vm = mon->vm;
+    virDomainObjPtr vm = NULL;
+    int retries = 0;
+    int monitor_fd;
 
     VIR_DEBUG("Monitor event loop thread starting");
 
+    while((monitor_fd = open(mon->monitorpath, O_RDONLY)) < 0) {
+        if ((errno == ENOMEM && ++retries < MONITOR_TMP_FAIL_RETRIES) ||
+                errno == EINTR) {
+            g_usleep(10000);    // 100 milli seconds
+            continue;
+        }
+        /*
+         * Any other error should be a BUG(kernel/libc/libvirtd)!
+         */
+        VIR_ERROR("Failed to open the monitor FIFO read end!");
```

Comment:
Do you think printing out the path for the `monitor_fd` would be useful for folks looking to debug the issue?

Reply:
Absolutely! Will add that to the the error message and probably one debug print before opening the FIFO as well. Thanks